rockcraft/extensions/flask.py

```diff
@@ -0,0 +1,169 @@
+        for property_name in properties:
+            if property_name in base_part and property_name not in new_part:
+                result[property_name] = base_part[property_name]
+            elif property_name not in base_part and property_name in new_part:
+                result[property_name] = new_part[property_name]
+            else:
+                result[property_name] = _apply_extension_property(
+                    base_part[property_name], new_part[property_name]
+                )
+        return result
+
+    def _merge_existing_part(self, part_name: str, part_def: dict) -> dict:
+        """Merge the new part with the existing part in the current rockcraft.yaml."""
+        existing_part = self.yaml_data.get("parts", {}).get(part_name, {})
+        return self._merge_part(existing_part, part_def)
+
+    def _gen_new_parts(self) -> Dict[str, Any]:
+        """Generate new parts for the flask extension.
+
+        Parts added:
+            - flask/dependencies: install Python dependencies
+            - flask/install-app: copy the flask project into the OCI image
+        """
+        if not (self.project_root / "requirements.txt").exists():
+            raise ValueError(
+                "missing requirements.txt file, "
+                "flask extension requires this file with flask specified as a dependency"
+            )
+        ignores = [".git", "node_modules", ".yarn"]
+        source_files = [
+            f
+            for f in os.listdir(self.project_root)
+            if f not in ignores and not f.endswith(".rock")
+        ]
+        renaming_map = {f: posixpath.join("srv/flask/app", f) for f in source_files}
+        install_app_part_name = "flask/install-app"
+        dependencies_part_name = "flask/dependencies"
+        # Users are required to compile any static assets prior to executing the
+        # rockcraft pack command, so assets can be included in the final OCI image.
+        install_app_part = {
+            "plugin": "dump",
+            "source": ".",
+            "organize": renaming_map,
+            "stage": list(renaming_map.values()),
+        }
+        dependencies_part = {
+            "plugin": "python",
+            "stage-packages": ["python3-venv"],
+            "source": ".",
+            "python-packages": ["gunicorn"],
+            "python-requirements": ["requirements.txt"],
+        }
+        snippet = {
+            dependencies_part_name: self._merge_existing_part(
+                dependencies_part_name, dependencies_part
+            ),
+            install_app_part_name: self._merge_existing_part(
+                install_app_part_name, install_app_part
+            ),
+        }
+        if self.yaml_data["base"] == "bare":
+            snippet["flask/container-processing"] = {
+                "plugin": "nil",
+                "source": ".",
+                "override-prime": "craftctl default\nmkdir -p tmp\nchmod 777 tmp",
```

Comment:
a bit surprised that this works, I guess `craftctl default` is changing the current dir to `CRAFT_PRIME`?
regardless, I think it's cleaner/more idiomatic to do something like `"override-build": "mkdir -m 777 ${CRAFT_PART_INSTALL}/tmp"`